### DISCUSSION TITLE
Storage: Type error in data.signedURL.

### DISCUSSION THREAD

**Discussion by ru88s** [supabase/supabase#1157] [Questions] [answered]
npm run build will result in a Type error.
What should I do?

![スクリーンショット 2021-04-15 18 23 46](https://user-images.githubusercontent.com/42270511/114847112-51942880-9e18-11eb-8058-163404a04c64.png)

This is suspicious
<img width="528" alt="スクリーンショット 2021-04-15 18 50 17" src="https://user-images.githubusercontent.com/42270511/114850173-74740c00-9e1b-11eb-8b76-82998272d92a.png">

It's my environment.
--Next.js
--Typescript
--Vercel

**Comment by inian** [comment, marked as answer]
Hi @ru88s, yes this was a error in types and I have just pushed out a fix. Can you try with supabase-js v1.11.4?
Also from your last discussion, it looked cleaner to access the signedURL from outside data too. So now you can do
```js
const { data, error, signedURL } = await supabase.storage.from("images").createSignedURL("cat.png", 60)
// you can use either of the following
console.log(data.signedURL);
console.log(signedURL);
```

**Reply by ru88s** [reply]
Hi @inian, Thank you for the improvement!
I'll try!

**Reply by ru88s** [reply]
This happened after the update, is it related?

![スクリーンショット 2021-04-15 19 21 57](https://user-images.githubusercontent.com/42270511/114854487-d9316580-9e1f-11eb-8bea-d209412364c0.png)

```
event - build page: /next/dist/pages/_error
wait  - compiling...
error - ./node_modules/@supabase/realtime-js/dist/module/RealtimeClient.js:11:0
Module not found: Can't resolve 'query-string'
```

**Reply by inian** [reply]
Can you try removing node_modules and try again? Cos `query-string` is no longer used in the realtime code base - https://github.com/supabase/realtime-js/commit/b1594210bf18707d7d4bac0d9c4e693e6a34eda0

**Reply by ru88s** [reply]
solved!
The signedURL is also easy to use.
Thank you so much!
